script.js

```diff
@@ -1,4 +1,608 @@
+  var chosenNum = 0;
+  if (choice == 1) {
+    chosenNum = Number(
+      String(rolls[(playerTurn - 1) * 3 + 1]) +
+        String(rolls[(playerTurn - 1) * 3 + 2])
+    );
+  } else if (choice == 2) {
+    chosenNum = Number(
+      String(rolls[(playerTurn - 1) * 3 + 2]) +
+        String(rolls[(playerTurn - 1) * 3 + 1])
+    );
+  }
+  return chosenNum;
+};
+
+// Function: Takes in Player X's turn, to return message on what numbers were rolled
+var playerXRolls = function (playerNum) {
+  var dice1Roll = playerRolls[(playerNum - 1) * 3 + 1];
+  var dice2Roll = playerRolls[(playerNum - 1) * 3 + 2];
+  var message = `You rolled <b>${dice1Roll}</b> for Dice 1 and <b>${dice2Roll}</b> for Dice 2.`;
+  return message;
+};
+
+// Function: Returns error message for not picking Dice 1 or 2
+var errorMessage = function () {
+  return `⛔️ Please try again. Enter '1' for Dice 1 or '2' for Dice 2.<br><br>${playerXRolls(
+    playerTurn
+  )}<br><br>Choose the order of the dice (1 or 2).`;
+};
+
+// Function: Generates Leaderboard for Standard & Variable Dice Versions (Highest Mode - Highest on top; Lowest Mode - Lowest on top)
+var leaderBoard = function () {
+  var score = `<br><br><b><u>Leaderboard</u></b> 🎯`;
+  var number = 11; // minimum number is 11 as a minimum of 2 dice is required
+  var ascendingOrderScores = []; // to store player number and scores in ascending order
+  while (ascendingOrderScores.length < scorePlayer.length) {
+    var innercounter = 0;
+    while (innercounter < scorePlayer.length) {
+      if (number == scorePlayer[innercounter]) {
+        ascendingOrderScores.push(scorePlayer[innercounter - 1]);
+        ascendingOrderScores.push(number);
+      }
+      innercounter += 1;
+    }
+    number += 1;
+  }
+  if (gameMode == "Highest Mode") {
+    while (ascendingOrderScores.length > 0) {
+      var xScore = ascendingOrderScores.pop();
+      var xPlayer = ascendingOrderScores.pop();
+      score += `<br>Player ${xPlayer}: <b>${xScore}</b>`; // "pop" scores and player numbers from the last element
+    }
+  } else if (gameMode == "Lowest Mode") {
+    var counter = 1;
+    while (counter - 1 < numOfPlayers) {
+      score += `<br>Player ${ascendingOrderScores[(counter - 1) * 2]}: <b>${
+        ascendingOrderScores[(counter - 1) * 2 + 1] // display score in existing ascendingOrderScores array
+      }</b>`;
+      counter += 1;
+    }
+  }
+  return score;
+};
+
+// Function: Game Phase 1 (Roll - Standard)
+var phase1 = function () {
+  playerRolls.push(String(playerTurn));
+  playerRolls.push(rollDice(6));
+  playerRolls.push(rollDice(6));
+  var message = `Welcome <b>Player ${playerTurn}</b>!<br><br>${playerXRolls(
+    playerTurn
+  )}`;
+  if (autoGenerate == 0) {
+    message += `<br><br>Choose the order of the dice (1 or 2).`;
+  } else if (autoGenerate == 1) {
+    message += `<br><b>Auto-Generate Mode</b>: Press '<b>Submit</b>' to continue...`;
+  }
+  gamePhase = 2;
+  return message;
+};
```

Comment:
'phase'... too much covid is it? like that autoGenerate above, applaudable laziness